### PR TITLE
add ability to reuse cookie from another session

### PR DESCRIPTION
**Pull request by kaloramik**
The Ingestion code currently calls Login on every task. We saw frequent failures 

 [20022] Additional login not permitted

We mitigated this before by reusing the cookie for search requests. Its unknown if this will completely solve this issue as we still call the Login url, but presumably this does not create additional sessions on their end because the cookie is preserved throughout the session.

### CODE REVIEW

**Comment by brianmartin** [comment]
@kaloramik Did you consider caching the login in the client itself (perhaps with a timeout)? If we create a client less frequently than the login rate limit, it could be a lot less headache on the caller of the client.

File: rets/http/client.py [hunk 1]
```diff
@@ -27,6 +27,7 @@ def __init__(self,
                  user_agent: str = 'rets-python/0.3',
```
Comment (brianmartin) [inline rets/http/client.py]: Obviously unrelated, but this should come from `rets.VERSION`

**Comment by brianmartin** [comment]
Ah disregard, I see the actual client is higher-level and each instance already caches the login.

**Comment by martinxsliu** [comment]
Can we verify that this works first? Its easy enough to log in, in one process and then use that cookie in another

**Comment by kaloramik** [comment]
it works, but the problem only manifests itself with a large number of logins. I could try to do a loop and compare. Actually I'll do that

**Comment by kaloramik** [comment]
Actually for loop didn't work, I had to use multi threading to repro an issue

```python
def login(limit):
    client = RetsClient(
    login_url=login_url,
    username=username,
    password=password,
    #cookie_str='ASP.NET_SessionId=coqlhnnrq21c0xlyyqvekbtf'
    )
    listing = client.get_resource('Property').get_class('Listing')
    r = listing.search(query='(Matrix_Unique_ID=0+)', limit=limit)
    return len(r.data)
```

```python
with ThreadPoolExecutor(max_workers=2) as executor:
    a = executor.submit(login, 1000)
    b = executor.submit(login, 2000)
    c = executor.submit(login, 3000)
    print(a.result())
    print(b.result())
    print(c.result())
```

This codes throws Too many outstanding queries (not the error I was expecting, but still an error)

While this code

```python
def login(limit):
    client = RetsClient(
    login_url=login_url,
    username=username,
    password=password,
    cookie_str='ASP.NET_SessionId=coqlhnnrq21c0xlyyqvekbtf'
    )
    listing = client.get_resource('Property').get_class('Listing')
    r = listing.search(query='(Matrix_Unique_ID=0+)', limit=limit)
    return len(r.data)
```

performs as expected

File: rets/http/client.py [hunk 2]
```diff
@@ -47,6 +48,15 @@ def __init__(self,
...
         self._session = None
 
+        # The user may provide an optional cookie_dict argument, which will be used on first login.
+        # When sending cookies (with a session_id) to the login url, the same cookie (session_id)
+        # is returned, which (most likely) means no additional login is created. On logout,
+        # this str is destroyed, and login will fetch a new cookies
+        self._cookie_dict = cookie_dict
+
+        # this session id is part of the rets standard for use with a user agent password
+        self._rets_session_id = ''
```
Comment (martinxsliu) [inline rets/http/client.py]: We'd need to rename `_session_id -> _rets_session_id` below.

Comment (kaloramik) [inline rets/http/client.py]: whoops, fixed

**Pull request merged**